### PR TITLE
1.0.5 : Fix connection drop on idle traffic

### PR DESCRIPTION
**Pull request by naveen-venkat**
* Add unit tests
* Bugfix: reconnecting when connection dropped

Connect to https://github.com/strongloop/loopback-connector-db2/issues/42

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

File: lib/ibmdb.js [hunk 1]
```diff
@@ -202,6 +202,10 @@ IBMDB.prototype.executeSQL = function(sql, params, options, callback) {
   var conn = self.connection;
   var stmt = {};
 
+  if (!self.connection.connected) {
```
Comment (0candy) [inline lib/ibmdb.js]: You could just use `if (!conn.connected)`

File: test/ibmdb.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    ibmdb.dataSource = ibmdb.datasource = {};
+    ibmdb.dataSource.connected = true;
+    ibmdb.dataSource.connecting = true;
+    done();
+});
+
+describe('IBMDB -> ', function () {
+  describe('executeSQL -> ', function () {
+    it('should set the datasource cnnected/connecting values to false', function (done) {
```
Comment (0candy) [inline test/ibmdb.test.js]: typo for `cnnected`

**Comment by 0candy** [comment]
@qpresley PTAL

**Comment by 0candy** [comment]
It would also help if you could put a more descriptive commit message and PR header like `Fix connection drop on idle traffic` or something like that so people know what this fix is for. Also, it would give a better description in the commit history and release notes when its published in the future.

**Comment by qpresley** [comment]
@0candy LGTM

**Comment by superkhau** [comment]
@naveen-venkat LGTM

**Comment by naveen-venkat** [comment]
@qpresley PTAL

**Comment by qpresley** [comment]
@naveen-venkat LGTM

**Comment by naveen-venkat** [comment]
Thanks @qpresley, when can we expect the release with this fix?

**Comment by 0candy** [comment]
@naveen-venkat Released version `1.0.5`

**Pull request merged**